### PR TITLE
Add Render Animation Worker Job

### LINKED ISSUE #7338: Export 3D render via worker job
Frontend:
- [x] build UI (probably a modal) to kick off job
- [x] Integrate pricing model
- [x] finish explanations texts
- [x] add header image / video
- [x] add validation for user inputs

Backend:
- [x] Validate frontend inputs

Worker
- [x] Finish Blender Animation Job. See https://github.com/scalableminds/voxelytics/pull/3336

### PR DESCRIPTION
**Pull request by fm3**
This PR adds the frontend and backend feature for starting a animation job of the current annotation / dataset.

image

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Start a worker with animation job (https://github.com/scalableminds/webknossos/issues/7338)
- Test in View mode and for an annotation
- For "Menu" in navbar select "Create Animation"
- Play around with available options to create various animations
- Restrict your orga to "Basic" plan and check plan enforcement

### Issues:
- fixes #7338

------
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [x] Removed dev-only changes like prints and application.conf edits
- [x] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)

### CODE REVIEW

File: frontend/javascripts/oxalis/view/action-bar/tracing_actions_view.tsx [hunk 1]
```diff
@@ -684,6 +690,18 @@ class TracingActionsView extends React.PureComponent<Props, State> {
     }
 
     menuItems.push(screenshotMenuItem);
+
+    if (activeUser?.isSuperUser) {
```
Comment (hotzenklotz) [inline frontend/javascripts/oxalis/view/action-bar/tracing_actions_view.tsx:694]: This is by design. Let's do a soft launch and try some larger datasets on the server first before releasing this publicly.

**Comment by hotzenklotz** [comment]
@philippotto I think this is ready for review. Feel free to only review the WK side. In case you setup up the worker too, perhaps you can do a rough review for that as well. Thanks.

I will create a page for the WK documentation later this week in a separate PR. Let's try to get this merged and soft launched for super users so that we can iron out any issue with the worker deployment and test performance in the real world.

**Comment by fm3** [comment]
I’d say the backend changes are straightforward enough that they don’t require a specialized review. @philippotto maybe you could quickly read through them as well while checking the front-end

File: frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+            />
+          </Col>
+          <Col span={16}>
+            <p>
+              Create a short, engaging animation of your data. Watch as the block of volumetric
+              image data shrinks to reveal segmented objects. Choose from three perspective options
+              and select the color layer and meshes you want to render.
+            </p>
+            <p>
+              For custom animations, please{" "}
```
Comment (philippotto) [inline frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx:253]: I'd elaborate here a bit (e.g., mention custom camera paths etc).

File: frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+    colorLayer: APIDataLayer,
+    selectedBoundingBox: BoundingBoxType,
+    meshSegmentIds: number[],
+  ) => {
+    //  Validate the select parameters and dataset to make sure it actually works and does not overload the server
+
+    const state = Store.getState();
+    const errorMessages: string[] = [];
+
+    const [_, estimated_texture_size] = selectMagForTextureCreation(
```
Comment (philippotto) [inline frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx:125]: please use camel case here

File: frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+
+    const isDtypeSupported = colorLayer.elementClass !== "uint24";
+    if (isDtypeSupported)
+      errorMessages.push("Sorry, animations are not supported for uInt24 datasets.");
+
+    const isDataset3D =
+      !is2dDataset(state.dataset) && (colorLayer.additionalAxes?.length || 0) === 0;
+    if (isDataset3D) errorMessages.push("Sorry, animations are only supported for 3D datasets.");
+
+    const isTooManyMeshes = meshSegmentIds.length > 30;
```
Comment (philippotto) [inline frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx:144]: please extract the 30 into a top level constant (also use that in line 147)

File: frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+  };
+
+  const submitJob = (evt: React.MouseEvent) => {
+    const state = Store.getState();
+    const boundingBox = userBoundingBoxes.find(
+      (bb) => bb.id === selectedBoundingBoxId,
+    )!.boundingBox;
+
+    // Submit currently visible pre-computed meshes
+    let meshSegmentIds = [] as number[];
```
Comment (philippotto) [inline frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx:165]:
```suggestion
    let meshSegmentIds: number[] = [];
```

File: frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx [hunk 6]
```diff
@@ -0,0 +1,383 @@
...
+        segmentationLayerName = visibleSegmentationLayer.name;
+      }
+    }
+
+    // Submit the configured min/max intensity info to support float datasets
+    const selectedColorLayer = getLayerByName(dataset, selectedColorLayerName);
+    const layerConfiguration = state.datasetConfiguration.layers[selectedColorLayer.name];
+    const { intensityRange } = layerConfiguration;
+    const defaultIntensityRange = getDefaultValueRangeOfLayer(dataset, selectedColorLayer.name);
+    const [intensityMin, intensityMax] = intensityRange || defaultIntensityRange;
```
Comment (philippotto) [inline frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx:193]: can you refactor this into a `getEffectiveIntensityRange(dataset, layerName, datasetConfiguration)` auxiliary accessor. Could be added to dataset_accessors.ts.

**Review by philippotto — CHANGES_REQUESTED** [review]
awesome stuff :+1: only left a couple of nitpicks.

also, one thing UX wise: I'd put the word "video" somewhere to clarify that the output medium is a video file. otherwise, it might be unclear to the user what they will get (a gif? a blender file? an upload to youtube?).

**Comment by philippotto** [comment]
> I’d say the backend changes are straightforward enough that they don’t require a specialized review. @philippotto maybe you could quickly read through them as well while checking the front-end

I did that yesterday, too, and everything looked good to me :)

**Comment by hotzenklotz** [comment]
@fm3 Did you built email notifications for this job as well?

**Comment by fm3** [comment]
I did not adapt the email code, no. My understanding is that a generic job completed email will be sent? Feel free to add a specialized email though :) Or I could do it as well if you tell me what it should say

File: frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx [hunk 7]
```diff
@@ -0,0 +1,383 @@
...
+  const submitJob = (evt: React.MouseEvent) => {
+    const state = Store.getState();
+    const boundingBox = userBoundingBoxes.find(
+      (bb) => bb.id === selectedBoundingBoxId,
+    )!.boundingBox;
+
+    // Submit currently visible pre-computed meshes
+    let meshSegmentIds = [] as number[];
+    let meshFileName = "meshfile.hdf5";
+    let segmentationLayerName = "segmentatation";
```
[the remark this answers is not in the document]
Comment (fm3) [reply frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx]: If this is optional then I’d say the backend should also understand that. I can probably add that today.

**Comment by hotzenklotz** [comment]
> I did not adapt the email code, no. My understanding is that a generic job completed email will be sent? Feel free to add a specialized email though :) Or I could do it as well if you tell me what it should say

I haven't tried the email notifications on my local dev setup (not configured). Let's start a soft launch and figure this out as well.

File: frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (fm3) [reply frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx]: I rechecked the backend code, and there the type is `segmentationLayerName: Option[String]`, so it should be possible to make this optional in the frontend already.

[the remark this answers is not in the document]
Comment (hotzenklotz) [reply frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx]: I switched the frontend values to `undefined` by default.

**Review by philippotto — APPROVED** [review]
Good stuff :+1:

**Comment by hotzenklotz** [comment]
Follow Up Issue: [#3350](https://github.com/scalableminds/voxelytics/issues/3350)

**Pull request merged**